### ISSUE TITLE
Combine all our READMEs into a single README

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#7795]
We should consider consolidating our READMEs into a single file with multiple sections -- a prospective user shouldn't have to figure out which README to read to find what they need, they should just be able to open the one-and-only README and look at the TOC and/or search for stuff. 

What do other folks think?

**Comment by mcourteaux** [comment]
I believe that this is a great idea. Perhaps one cool thing would be to auto-generate the consolidated README using a recipe of other READMEs (like a `#include "src/irvisualizer/README.md"`, but more flexible). This would allow us to reference certain blocks of in-code documentation, or `enum` declarations.

One example of such a piece of documentation that is currently only available in-code that I think should be in a README is the list of Target Features: https://halide-lang.org/docs/_halide_runtime_8h.html#a2ccb96b3d427fff8f1d68cc5f1e92f3a It took me really long to even find this, even with the search in Doxygen. (There is also this one, which is not helpful: https://halide-lang.org/docs/struct_halide_1_1_target.html#a9398276edd5784185e1a08f8a4dfc43f)

I don't know if there are good tools for this.

**Comment by steven-johnson** [comment]
> the list of Target Features

Agreed that it should be more visible -- but we almost certainly want it to be driven from Doxygen comments (or similar) to keep things up to date. (This presumably implies some tool-driven approach to build the README file, or at least verify it's up to date via presubmit checks.)

(Also: marking as `contributor project` because this seems like something that would be easy for someone outside the core codebase to look at.)
